application/models/services/SurveyUpdater/GeneralSettings.php

```diff
@@ -0,0 +1,570 @@
+<?php
+
+namespace LimeSurvey\Models\Services\SurveyUpdater;
+
+use Survey;
+use Permission;
+use LSYii_Application;
+use PluginEvent;
+use Date_Time_Converter;
+use LimeSurvey\SessionData;
+use LimeSurvey\PluginManager\PluginManager;
+use LimeSurvey\Models\Services\Exception\{
+    ExceptionPersistError,
+    ExceptionNotFound,
+    ExceptionPermissionDenied
+};
+
+/**
+ * Service GeneralSettings
+ *
+ * Service class for survey language setting updating.
+ *
+ * Dependencies are injected to enable mocking.
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ */
+class GeneralSettings
+{
+    private Permission $modelPermission;
+    private Survey $modelSurvey;
+    private LSYii_Application $yiiApp;
+    private SessionData $sessionData;
+    private PluginManager $yiiPluginManager;
+    private LanguageConsistency $languageConsistency;
+
+    const FIELD_TYPE_YN = 'yersno';
+    const FIELD_TYPE_DATETIME = 'dateime';
+    const FIELD_TYPE_GAKEY = 'gakey';
+    const FIELD_TYPE_USE_CAPTCHA = 'use_captcha';
+
+    const GA_GLOBAL_KEY = '9999useGlobal9999';
+
+    public function __construct(
+        Permission $modelPermission,
+        Survey $modelSurvey,
+        LSYii_Application $yiiApp,
```

Comment:
Or just `$app`? Only used to load libraries, so maybe just load them before instead? Unless that's "for later", but then we might want to write them down somehow, either as "todo" or a task. Or both.

Reply:
Load them before where?

Reply:
Load them before the GeneralSettings object is constructed?

Meh, lack of autoloading for functions in PHP is a PITA...

Reply:
Btw, I'm assuming controller actions will only be tested with functional (scripting the browser) testing, not unit tests. At least in our current architecture.

Reply:
The gain is to have one less injection and to not pollute service classes with "administrative code". The cost is some code duplication, but that's only due to lack of autoloading of functions in PHP.

At least you could import the needed libs in the constructor, no need to assign app to a class property in this case.